src/hotspot/share/runtime/thread.cpp

```diff
@@ -1644,19 +1639,13 @@ void JavaThread::check_and_handle_async_exceptions() {
       // We might have blocked in a ThreadBlockInVM wrapper in the call above so make sure we process pending
       // suspend requests and object reallocation operations if any since we might be going to Java after this.
       SafepointMechanism::process_if_requested_with_exit_check(this, true /* check asyncs */);
-      return;
-    }
-    case _thread_in_native: {
-      ThreadInVMfromNative tiv(this);
-      JavaThread* THREAD = this;
-      Exceptions::throw_unsafe_access_internal_error(THREAD, __FILE__, __LINE__, "a fault occurred in an unsafe memory access operation");
-      return;
+      break;
```

Comment:
Other than now including the final assertion to be executed, I don't see why we need to change from "return" to "break". That final assertion seems unnecessary as it basically checks that Exceptions::throw... actually threw. I'd be tempted to just delete the final assertion and keep the return statements. But not a major concern either way.